lms/djangoapps/program_enrollments/api/v1/views.py

```diff
@@ -0,0 +1,249 @@
+# -*- coding: utf-8 -*-
+"""
+ProgramEnrollment Views
+"""
+from __future__ import unicode_literals
+from django.http import Http404
+from django.http import HttpResponse
+from course_modes.models import CourseMode
+from student.models import CourseEnrollment
+from opaque_keys.edx.keys import CourseKey
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.exceptions import ValidationError
+from rest_framework.views import APIView
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.status import (
+    HTTP_207_MULTI_STATUS,
+    HTTP_413_REQUEST_ENTITY_TOO_LARGE,
+    HTTP_422_UNPROCESSABLE_ENTITY,
+)
+
+from openedx.core.djangoapps.catalog.utils import get_programs
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+from lms.djangoapps.program_enrollments.api.v1.constants import CourseEnrollmentResponseStatuses
+from lms.djangoapps.program_enrollments.models import ProgramEnrollment, ProgramCourseEnrollment
+from lms.djangoapps.program_enrollments.serializers import ProgramCourseEnrollmentRequestSerializer
+
+
+class ProgramEnrollmentsView(APIView):
+    """
+    POST view for ProgramEnrollments
+    """
+
+    def get(self, request, *args, **kwargs):
+        return HttpResponse('result')
+
+
+class ProgramSpecificViewMixin(object):
+    """
+    A mixin for views that operate on or within a specific program.
+    """
+    def __init__(self, *args, **kwargs):
+        super(ProgramSpecificViewMixin, self).__init__(*args, **kwargs)
+        self._program = None
+
+    @property
+    def program(self):
+        """
+        The program specified by the `program_uuid` URL parameter.
+        """
+        if self._program is None:
+            program = get_programs(uuid=self.kwargs['program_uuid'])
+            if program is None:
+                raise Http404()
+            self._program = program
+        return self._program
+
+
+class ProgramCourseRunSpecificViewMixin(ProgramSpecificViewMixin):
+    """
+    A mixin for views that operate on or within a specific course run in a program
+    """
+    def __init__(self, *args, **kwargs):
+        super(ProgramCourseRunSpecificViewMixin, self).__init__(*args, **kwargs)
+        self._course_run = None
+        self._course_key = None
+
+    def check_existence_and_membership(self):
```

Comment:
Maybe call this `check_course_existence_and_membership`? Inside the mixin it's obvious what the method refers to, but in other contexts, not so much.